netdeploy/network.go

```diff
@@ -57,25 +58,37 @@ type Network struct {
 
 // CreateNetworkFromTemplate uses the specified template to deploy a new private network
 // under the specified root directory.
-func CreateNetworkFromTemplate(name, rootDir, templateFile, binDir string, importKeys bool, nodeExitCallback nodecontrol.AlgodExitErrorCallback, consensus config.ConsensusProtocols, overrideDevMode bool) (Network, error) {
+func CreateNetworkFromTemplate(name, rootDir string, templateReader io.Reader, binDir string, importKeys bool, nodeExitCallback nodecontrol.AlgodExitErrorCallback, consensus config.ConsensusProtocols, overrideDevMode bool) (Network, error) {
 	n := Network{
 		rootDir:          rootDir,
 		nodeExitCallback: nodeExitCallback,
 	}
 	n.cfg.Name = name
-	n.cfg.TemplateFile = templateFile
 
-	template, err := loadTemplate(templateFile)
+	var err error
+	var template NetworkTemplate
```

Comment:
The reward pool bug is due to missing template defaults.

The failing test has this at the top:
>	fixture.Setup(t, filepath.Join("nettemplates", "FourNodes.json"))

Taking a look at FourNodes.json, it does not set "RewardsPoolBalance". To see what happens, I go ahead and add a value:
> "RewardsPoolBalance": 125000000000000,

The test passes!

So why are the defaults missing? The code is no skipping `loadTemplate`, since we can now call `loadTemplateFromReader` directly. It looked a little strange that the template variable needs to be defined ahead of time, so I looked at how `loadTemplate` did the initialization and found the missing piece:

```suggestion
	template := defaultNetworkTemplate
```

(this is not your defaultNetworkTemplate variable, `netdeploy` has its own version which sets a few values -- including the RewardsPoolBalance)

Reply:
Wow, embarrassed I didn't see that. Thanks.